blockchain/chain.go

```diff
@@ -307,48 +307,58 @@ type StakeVersions struct {
 // GetStakeVersions returns a cooked array of StakeVersions.  We do this in
 // order to not bloat memory by returning raw blocks.
 func (b *BlockChain) GetStakeVersions(hash *chainhash.Hash, count int32) ([]StakeVersions, error) {
-	// Nothing to do if no count requested.
-	if count == 0 {
-		return nil, nil
-	}
-
-	if count < 0 {
-		return nil, fmt.Errorf("count must not be less than zero - "+
-			"got %d", count)
-	}
+	exists, err := b.HaveBlock(hash)
 
-	b.chainLock.Lock()
-	defer b.chainLock.Unlock()
-
-	if count > int32(b.bestNode.height) {
-		count = int32(b.bestNode.height)
-	}
-
-	startNode, err := b.findNode(hash, 0)
 	if err != nil {
 		return nil, err
 	}
 
-	result := make([]StakeVersions, 0, count)
-	prevNode := startNode
-	for i := int32(0); prevNode != nil && i < count; i++ {
-		sv := StakeVersions{
-			Hash:         prevNode.hash,
-			Height:       prevNode.height,
-			BlockVersion: prevNode.header.Version,
-			StakeVersion: prevNode.header.StakeVersion,
-			Votes:        prevNode.votes,
+	if exists {
```

Comment:
Yes, we prefer to have the non-error code paths from top to bottom of the function.  Return early for errors.